packages/ag-charts-community/src/scene/node.ts

```diff
@@ -423,32 +403,17 @@ export abstract class Node<D = any> {
         this.markDirty(property);
     }
 
-    markDirty(property?: string) {
-        const { _dirty } = this;
+    markDirtyZIndex() {
+        this.parentNode?.markDirtyZIndex();
+    }
 
+    markDirty(property?: string) {
         if (property != null && this._debugDirtyProperties) {
             this.markDebugProperties(property);
         }
 
-        const noParentCachedBBox = this.cachedBBox == null;
-        if (noParentCachedBBox && _dirty) return;
-
-        this.invalidateCachedBBox();
-        this._dirty = true;
-        if (this.parentNode) {
-            this.parentNode.markDirty();
-        }
-    }
-
-    markClean() {
-        if (!this._dirty) return;
-
-        this._dirty = false;
-        this.debugDirtyProperties();
-
-        for (const child of this.children()) {
-            child.markClean();
-        }
+        this.cachedBBox = undefined;
+        this.parentNode?.markDirty();
```

Comment:
The original intent was to avoid percolating `markDirty()` calls upwards if this has happened previously, so we need to be careful that we haven't reduced performance in other cases by causing a lot more calls upwards.

Reply:
I can't figure out how to do this. I tried not doing it, but it breaks tests. There's probably more performance to be won here - but it's at least not a regression